syntax = "proto3";

package user;

service UserService {
    rpc AddUser (UserRequest) returns (UserResponse);
    rpc GetUser (UserId) returns (UserResponse);
    rpc UpdateUser (UserRequest) returns (UserResponse);
}

message UserRequest {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}

message UserId {
    int32 id = 1;
}

message UserResponse {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}
